scripts/cmake/xtensa-toolchain.cmake

```diff
@@ -45,3 +45,20 @@ set(CMAKE_FIND_ROOT_PATH  ".")
 set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
 set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
 set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+
+if(XCC)
+	# get compiler description
+	execute_process(
+		COMMAND ${CMAKE_C_COMPILER} --show-config=config
+		OUTPUT_VARIABLE cc_config_output
+		OUTPUT_STRIP_TRAILING_WHITESPACE
+		ERROR_QUIET
+	)
+
+	string(REGEX MATCH "[a-zA-Z]+-[0-9]+.[0-9]+-[a-zA-Z]*" XCC_TOOLS_VERSION "${cc_config_output}")
+	if(NOT XCC_TOOLS_VERSION)
+		message(WARNING "Couldn't get compiler description from '${XCC_TOOLS_VERSION}'")
+	endif()
+else()
+	set(XCC_TOOLS_VERSION "")
```

Comment:
@ktrzcinx I see it's already in other fields in code so no point in duplicating it here

Reply:
After second thought actually you can add 1 thing here - `${TOOLCHAIN}`, preferably last path component, so if someone passes here `/path/to/xtensa-cnl-elf`, then it will ve `xtensa-cnl-elf`.

It may help in case if someone tries to compile different binary in the previoiusly configured cmake folder for another platform and somehow failed setup. Usually you'd have another errors along the way in that scenario, but it still may happen to build if someone is taking some weird modular approach (f.e. plays with 3rd party static libs).

Reply:
done